### ISSUE TITLE
Optimization for Block Entity / Entity Rendering

### ISSUE THREAD

**Issue by Niknokinater** [jellysquid3/Sodium#1213]
**Is your feature request related to a problem? Please describe.**
Entity / block entity visual lag. There is a vanilla issue where block entities & entities take up an incredible amount of resources just by looking at them and this destroys performance in some of the handiest places, such as animals pens or chest rooms. To my knowledge, this affects signs, item frames, some workstation blocks, chests (as noted above)- even beds.

[Scenario: i7-7700 @ 4.1GHz, RTX 3070Ti, 32Gb of RAM | Render & Sim distance are lowest feasible at 10 & 5 respectively. Any lower on render distance and mob spawning breaks as described in MC-2536. First FPS listing is looking at a jungle in the opposite direction of chest room. Second is looking at the chest room].

For example, whilst vanilla can drop from ~130 to ~80 (negligible at a goal of 60fps). 

In Sodium the fps drops from ~300 to ~150 under the same circumstances (also negligible, aiming for 60fps).
Throw Iris into the mix and the game drops from ~130 to ~20, coinciding with an outright GPU utilization drop of 70%.

**Describe the solution you'd like**
Simply put, optimization for entities & block entities. It's rather frustrating that they impact framerate so heavily so as to affect the usage of perfectly available resources.

**Comment by muzikbike** [comment]
Isn't this what Enchanced Block Entities does?

**Comment by Niknokinater** [comment]
> Isn't this what Enchanced Block Entities does?

Having just researched that (thanks for relaying that info !), it appears as if its goal is the same, yes. It seems like Enchanted Block Items is still on base 1.18, unfortunately, whilst depending on both Indium & the Fabric API (the latter being in the sightline for native Sodium support someday, if I'm reading this correctly, making the former unnecessary when it arrives).

This kind of optimization is very much in-line with Sodium's purpose and it would make Sodium that much more impactful.
